pkg/ebpf/controlplane/processes.go

```diff
@@ -1,128 +1,76 @@
+	parentTid, err := parse.ArgVal[int32](args, "up_parent_tid")
+	errs = append(errs, err)
+	parentNsTid, err := parse.ArgVal[int32](args, "up_parent_ns_tid")
+	errs = append(errs, err)
+	parentPid, err := parse.ArgVal[int32](args, "up_parent_pid")
+	errs = append(errs, err)
+	parentNsPid, err := parse.ArgVal[int32](args, "up_parent_ns_pid")
+	errs = append(errs, err)
+	parentStartTime, err := parse.ArgVal[uint64](args, "up_parent_start_time")
+	errs = append(errs, err)
 
 	// Thread Group Leader (might be the same as the "child", if "child" is a process)
-
 	leaderTid, err := parse.ArgVal[int32](args, "leader_tid")
-	if err != nil {
-		return errfmt.Errorf("error parsing leader_tid: %v", err)
-	}
+	errs = append(errs, err)
 	leaderNsTid, err := parse.ArgVal[int32](args, "leader_ns_tid")
-	if err != nil {
-		return errfmt.Errorf("error parsing leader_ns_tid: %v", err)
-	}
+	errs = append(errs, err)
 	leaderPid, err := parse.ArgVal[int32](args, "leader_pid")
-	if err != nil {
-		return errfmt.Errorf("error parsing leader_pid: %v", err)
-	}
+	errs = append(errs, err)
 	leaderNsPid, err := parse.ArgVal[int32](args, "leader_ns_pid")
-	if err != nil {
-		return errfmt.Errorf("error parsing leader_ns_pid: %v", err)
-	}
+	errs = append(errs, err)
 	leaderStartTime, err := parse.ArgVal[uint64](args, "leader_start_time")
-	if err != nil {
-		return errfmt.Errorf("error parsing leader_start_time: %v", err)
-	}
+	errs = append(errs, err)
 
 	// Child (might be a process or a thread)
-
 	childTid, err := parse.ArgVal[int32](args, "child_tid")
-	if err != nil {
-		return errfmt.Errorf("error parsing child_tid: %v", err)
-	}
+	errs = append(errs, err)
 	childNsTid, err := parse.ArgVal[int32](args, "child_ns_tid")
-	if err != nil {
-		return errfmt.Errorf("error parsing child_ns_tid: %v", err)
-	}
+	errs = append(errs, err)
 	childPid, err := parse.ArgVal[int32](args, "child_pid")
-	if err != nil {
-		return errfmt.Errorf("error parsing child_pid: %v", err)
-	}
+	errs = append(errs, err)
 	childNsPid, err := parse.ArgVal[int32](args, "child_ns_pid")
-	if err != nil {
-		return errfmt.Errorf("error parsing child_ns_pid: %v", err)
-	}
-	childStartTime, err := parse.ArgVal[uint64](args, "child_start_time")
-	if err != nil {
-		return errfmt.Errorf("error parsing child_start_time: %v", err)
-	}
-
-	// Sanity check: check if eBPF Hash matches userland Hash
-
-	taskHashVerifier := utils.HashTaskID(uint32(childTid), childStartTime)
-	parentHashVerifier := utils.HashTaskID(uint32(parentTid), parentStartTime)
-	leaderHashVerifier := utils.HashTaskID(uint32(leaderTid), leaderStartTime)
+	errs = append(errs, err)
+	childStartTime, err := parse.ArgVal[uint64](args, "start_time") // child_start_time
+	errs = append(errs, err)
 
-	if childHash != taskHashVerifier ||
-		parentHash != parentHashVerifier ||
-		leaderHash != leaderHashVerifier {
-		return errfmt.Errorf("eBPF Hash does not match")
+	// Handle errors
+	for _, err := range errs {
```

Comment:
Hmm, is it not preferred to just return the error when you get it?
I don't see any pros to this approach.
If you wanted to feed the tree anyways that it would have made sense...

Reply:
readability (1 line in between the calls only)